src/Microsoft.DotNet.TemplateLocator/TemplateLocator.cs

```diff
@@ -5,68 +5,122 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
-using NuGet.Common;
-using NuGet.Protocol;
+using Microsoft.DotNet.DotNetSdkResolver;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
 
 namespace Microsoft.DotNet.TemplateLocator
 {
     public sealed class TemplateLocator
     {
-        private DirectoryInfo _dotnetSdkTemplatesLocation;
+        private readonly Lazy<NETCoreSdkResolver> _netCoreSdkResolver;
 
         public TemplateLocator()
+            : this(Environment.GetEnvironmentVariable, VSSettings.Ambient)
         {
-            string mockTemplateLocation = Environment.GetEnvironmentVariable("MOCKDOTNETSDKTEMPLATESLOCATION");
-            if (!string.IsNullOrWhiteSpace(mockTemplateLocation))
-            {
-                _dotnetSdkTemplatesLocation = new DirectoryInfo(mockTemplateLocation);
-            }
         }
 
-        public IReadOnlyCollection<IOptionalSdkTemplatePackageInfo> GetDotnetSdkTemplatePackages(string sdkVersion)
+        /// <summary>
+        /// Test constructor
+        /// </summary>
+        public TemplateLocator(Func<string, string> getEnvironmentVariable, VSSettings vsSettings)
+        {
+            _netCoreSdkResolver =
+                new Lazy<NETCoreSdkResolver>(() => new NETCoreSdkResolver(getEnvironmentVariable, vsSettings));
+        }
+
+        public IReadOnlyCollection<IOptionalSdkTemplatePackageInfo> GetDotnetSdkTemplatePackages(
```

Comment:
What is the behavior supposed to be for workloads that are not installed?  The template packs would be listed in the manifest, but the packages would not be found on disk.  Should this method be filtering them out?

Reply:
No. That is a good point. Although I think the manifest file reader should do the filtering.

Reply:
In the case of resolving an SDK, we don't want the manifest reader to do the filtering.  The SDK resolver filters if the SDK pack is not installed, and adds the items that tell the build to fail and VS that the workload is required to build the project.